### PR TITLE
Add Overlays for Argo on OCP4 with slightly modified role definitions

### PR DESCRIPTION
**Pull request by gmfrasca**

### CODE REVIEW

File: argo/overlays/ocp4.dh-prod-argo/deployments/argo-server_patch.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: argo-server
+spec:
+  template:
+    spec:
+      nodeSelector:
+        $patch: replace
+        beta.kubernetes.io/os: linux
```
Comment (lucferbux) [inline argo/overlays/ocp4.dh-prod-argo/deployments/argo-server_patch.yaml:11]: New line

File: argo/overlays/ocp4.dh-prod-argo/deployments/workflow-controller_patch.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: workflow-controller
+spec:
+  template:
+    spec:
+      nodeSelector:
+        $patch: replace
+        beta.kubernetes.io/os: linux
```
Comment (lucferbux) [inline argo/overlays/ocp4.dh-prod-argo/deployments/workflow-controller_patch.yaml:11]: New line

**Comment by lucferbux** [comment]
It looks good to me, but I would need to review the role definitions. A part from that it needs new lines in some of the files.

**Review by lucferbux — CHANGES_REQUESTED** [review]

**Review by anishasthana — COMMENTED** [review]
/lgtm

**Comment by gmfrasca** [comment]
Just to comment, I've found that `kustomize build` was failing for this application even before/without the contents of this PR. specifically, it appears to not like the inclusion of a resource from raw.githubusercontent.com as kustomize only allows very specific URL domains for github.

**Review by anishasthana — COMMENTED** [review]
/lgtm

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**